ql/src/java/org/apache/hadoop/hive/ql/ddl/table/partition/drop/AbstractDropPartitionAnalyzer.java

```diff
@@ -103,34 +105,40 @@ protected void analyzeCommand(TableName tableName, Map<String, String> partition
     re.noLockNeeded();
     inputs.add(re);
 
-    addTableDropPartsOutputs(table, partitionSpecs.values(), !ifExists);
-
+    boolean dropPartUseBase = HiveConf.getBoolVar(conf, ConfVars.HIVE_ACID_DROP_PARTITION_USE_BASE)
+        || HiveConf.getBoolVar(conf, ConfVars.HIVE_ACID_LOCKLESS_READS_ENABLED)
+      && AcidUtils.isTransactionalTable(table);
+    
+    addTableDropPartsOutputs(table, partitionSpecs.values(), !ifExists, dropPartUseBase);
+    
     AlterTableDropPartitionDesc desc =
-        new AlterTableDropPartitionDesc(tableName, partitionSpecs, mustPurge, replicationSpec);
-    rootTasks.add(TaskFactory.get(new DDLWork(getInputs(), getOutputs(), desc)));
-
-    postProcess(tableName, table, desc);
```

Comment:
Removed the `postProcess`. Is it intentional?

Reply:
yes, it's not needed, both overridden methods have the same functionality
````
    if (desc.mayNeedWriteId()) {
      setAcidDdlDesc(desc);
    }
````